_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+```
+
+Create a `.env` file in the project root to store the Neo4j credentials:
+
+__.env__
+```shell
+export SPRING_NEO4J_AUTHENTICATION_PASSWORD=verysecret
+```
+
+Download the following seed files to some folder:
+
+- [CompanyDataAmericans](https://guides.neo4j.com/ukcompanies/data/CompanyDataAmericans.csv)
+- [LandOwnershipAmericans](https://guides.neo4j.com/ukcompanies/data/LandOwnershipAmericans.csv)
+- [PSCAmericans.csv](https://guides.neo4j.com/ukcompanies/data/PSCAmericans.csv)
+
+Create the folder `src/main/docker` and create a file `neo4j.yml` there, with the following content:
+
+__neo4j.yml__
+```yml
+# This configuration is intended for development purpose, it's **your** responsibility to harden it for production
+name: companies
+services:
+  neo4j:
+    image: neo4j:5
+    volumes:
+      - <csv-folder>:/var/lib/neo4j/import
+    environment:
+      - NEO4J_AUTH=neo4j/${NEO4J_PASSWORD}
+      - NEO4JLABS_PLUGINS=["apoc"]
+    # If you want to expose these ports outside your dev PC,
+    # remove the "127.0.0.1:" prefix
+    ports:
+      - '127.0.0.1:7474:7474'
+      - '127.0.0.1:7687:7687'
+    healthcheck:
+      test: ['CMD', 'wget', 'http://localhost:7474/', '-O', '-']
+      interval: 5s
+      timeout: 5s
+      retries: 10
+```
+
+Create the file `src/main/docker/.env` with the following content:
+
+__.env__
+```shell
+NEO4J_PASSWORD=verysecret
+```
+
+As you can see the compose file will mount `<csv-folder>` to a `/var/lib/neo4j/import` volume, making the content accessible from the running neo4j container.
+Replace `<csv-folder>` with the path to the CSV files downloaded before.
+
+In a terminal, go to the `docker` folder and run:
+
+```shell
+docker compose -f neo4j.yml up
+```
+
+### Run the API server
+
+Go to the project root folder and start the application with:
+
+```shell
+source .env && ./gradlew bootRun
+```
+
+Wait for the logs to inform the seed data migrations have run (it might take a while):
+
+```
+2023-08-02T13:06:14.386-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 001 ("Constraint").
+2023-08-02T13:06:23.379-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 002 ("Company").
+2023-08-02T13:11:23.693-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 003 ("Person").
+2023-08-02T13:21:03.680-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 004 ("PersonCompany").
+2023-08-02T13:21:06.519-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 005 ("CompanyData").
+2023-08-02T13:21:06.551-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 006 ("Land").
+```
+
+Test the API with GraphiQL at **http://localhost:8080/graphiql**. In the query box on the left, paste the following query
```

Comment:
```suggestion
Test the API with GraphiQL at `http://localhost:8080/graphiql`. In the query box on the left, paste the following query:
```
Just following the convention we've used for localhost URLs over the years. I'm not sure that the Auth0 cares as much as the OktaDev blog. I'm just trying to be consistent.